client-java-contrib/generate.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script orchestrates the code generation procedure. It is executed inside a crdgen
+# container in order to minimize the environment dependencies on the host, being Docker only.
+
+CRD_URL=${CRD_URL:-}
+OUTPUT_DIR=${OUTPUT_DIR:-}
+KUBERNETES_CRD_GROUP_PREFIX=${KUBERNETES_CRD_GROUP_PREFIX:-}
+PACKAGE_NAME=${PACKAGE_NAME:-}
+
+print_usage() {
+  echo "Usage: generate Java model classes from CRDs" >& 2
+  echo " -n: the prefix of the target CRD's api group to generate." >& 2
+  echo " -p: the base package name of the generated java project. " >& 2
+  echo " -o: output directory of the generated java project. " >& 2
+  echo " -u: url location of the YAML manifest to install CRDs to a Kubernetes cluster. " >& 2
+}
+
+while getopts 'u:n:p:o:' flag; do
+  case "${flag}" in
+    u) CRD_URL="${OPTARG}" ;;
+    n) KUBERNETES_CRD_GROUP_PREFIX="${OPTARG}" ;;
+    p) PACKAGE_NAME="${OPTARG}" ;;
+    o) OUTPUT_DIR="${OPTARG}" ;;
+    *) print_usage
+       exit 1 ;;
+  esac
+done
+
+# create a KinD cluster on the host
+kind create cluster
+
+# install CRDs to the KinD cluster and dump the swagger spec
+kubectl apply -f "${CRD_URL}"
+sleep 5
+kubectl get --raw="/openapi/v2" > /tmp/swagger
```

Comment:
```suggestion
kubectl get --raw="/openapi/v2" > /tmp/swagger

echo "Verifying CRD installation.."
kubectl get crd -o name \
  | while read L
    do
      if [[ $(kubectl get $L -o jsonpath='{.status.conditions[?(@.type=="NonStructuralSchema")].status}') == "True" ]]; then
        echo "$L failed publishing openapi schema because it's attached non-structral-schema condition."
        exit 1
      fi
      if [[ $(kubectl get $L -o jsonpath='{.preserveUnknownFields}') == "true" ]]; then
        echo "$L failed publishing openapi schema because it explicitly disabled unknown fields pruning."
        exit 1
      fi
      echo "$L successfully installed"
    done
```

adding a post crd installation check in case generating empty models.

Reply:
Thanks for contributing this validation logic in shell script, will incorporate it.